src/sentry/integrations/vercel/integration.py

```diff
@@ -76,6 +77,85 @@ def get_organization_config(self):
 
         return fields
 
+    def update_organization_config(self, data):
+        # data = {"project_mappings": [[sentry_project_id, vercel_project_id]]}
+
+        metadata = self.model.metadata
+        vercel_client = VercelClient(metadata["access_token"], metadata.get("team_id"))
+        config = self.org_integration.config
+        sentry_project_id = data["project_mappings"][-1][0]
+        vercel_project_id = data["project_mappings"][-1][1]
+        sentry_project = Project.objects.get(id=sentry_project_id)
+        sentry_project_dsns = ProjectKey.objects.filter(project=sentry_project).all()
+        enabled_dsns = [dsn for dsn in sentry_project_dsns if dsn.is_active]
+        if enabled_dsns:  # in case there are no enabled DSNs
+            sentry_project_dsn = enabled_dsns[0].get_dsn(public=True)
+        else:
+            raise
+        try:
+            self.create_secret(
+                vercel_client, vercel_project_id, "SENTRY_ORG", sentry_project.organization.slug
+            )
+        except Exception:
+            raise
+
+        try:
+            self.create_secret(
+                vercel_client, vercel_project_id, "SENTRY_PROJECT", sentry_project.slug
+            )
+        except Exception:
+            raise
+
+        try:
+            self.create_secret(
+                vercel_client, vercel_project_id, "NEXT_PUBLIC_SENTRY_DSN", sentry_project_dsn
+            )
+        except Exception:
+            raise
+
+        config.update(data)
+        self.org_integration.update(config=config)
+
+    def get_env_vars(self, client, vercel_project_id):
+        return client.get(path=client.GET_ENV_VAR_URL % vercel_project_id)
+
+    def get_secret(self, client, name):
+        try:
+            exists = True
+            client.get(path=client.GET_SECRET_URL % name)
+        except ApiError as e:
+            if e.code == 404:
+                exists = False
+            else:
+                raise
+        return exists
+
+    def env_var_already_exists(self, client, vercel_project_id, name):
+        return len(
+            filter(
+                lambda env_var: env_var["key"] == name,
+                self.get_env_vars(client, vercel_project_id)["envs"],
+            )
+        )
+
+    def create_secret(self, client, vercel_project_id, name, value):
+        secret = None
+        if not self.get_secret(client, name):
+            secret = client.post(path=client.SECRETS_URL, data={"name": name, "value": value})[
+                "uid"
+            ]
+        if secret is not None and not self.env_var_already_exists(client, vercel_project_id, name):
```

Comment:
@ceorourke If the secret already exists, why would that stop us from making the environment variable for that project?

Reply:
@scefali This is checking if we have a secret to use and that the env var already exists or not, not the existence of the secret. If the env var already exists it will error attempting to create another one with the same name.

Reply:
@ceorourke What I am trying to say is in the case where the secret already exists, the env variable for that project might not. I think we should check to see if the environment variable exists for that project even if the secret already exists.